wgpu-hal/src/gles/egl.rs

```diff
@@ -532,12 +532,42 @@ impl Inner {
                 .query_string(Some(display), khronos_egl::CLIENT_APIS)
                 .unwrap()
                 .to_string_lossy();
-            client_apis
+            if client_apis
                 .split(' ')
                 .any(|client_api| client_api == "OpenGL")
+            {
+                // check if GL version is supported
+                egl.bind_api(khronos_egl::OPENGL_API).unwrap();
+                let ctx = egl
+                    .create_context(display, config, None, &[khronos_egl::NONE])
+                    .map_err(|e| {
+                        crate::InstanceError::with_source(
+                            String::from("unable to create GL context"),
+                            e,
+                        )
+                    })?;
+                let _ = egl.make_current(display, None, None, Some(ctx));
+                unsafe {
```

Comment:
that unsafe scope is larger than it has to be, please shrink it down to the bare minimum

Reply:
I've cut down the unsafe scope